lib/util/my_gzip.ml

```diff
@@ -1,102 +1,48 @@
-type in_channel = {
-  ic : Stdlib.in_channel;
-  mutable decoder : Gz.Inf.decoder;
-  i : De.bigstring;
-  o : De.bigstring;
-  tmp : Bytes.t;
-  lines : string Queue.t;
-  tail : Buffer.t;
-}
+type in_channel = { content : string; mutable pos : int }
 
-let open_in path =
+let gunzip_file path =
   let ic = Stdlib.open_in_bin path in
+  Fun.protect ~finally:(fun () -> Stdlib.close_in ic) @@ fun () ->
```

Comment:
This is not correct. You must not call `close_in` in a finalizer! You have to call `close_in_noerr`.